packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -769,3 +769,78 @@ class ImportedCluster extends Resource implements ICluster {
     }
   }
 }
+
+/**
+ * Properties for EksOptimizedAmi
+ */
+export interface EksOptimizedAmiProps {
+  /**
+   * What instance type to retrieve the image for (normal or GPU-optimized)
+   *
+   * @default Normal
+   */
+  readonly nodeType?: NodeType;
+
+  /**
+   * The Kubernetes version to use
+   *
+   * @default The latest version
+   */
+  readonly kubernetesVersion?: string;
+}
+
+/**
+ * Construct an Amazon Linux 2 image from the latest EKS Optimized AMI published in SSM
+ */
+export class EksOptimizedImage implements ec2.IMachineImage {
+  private readonly nodeType?: NodeType;
+  private readonly kubernetesVersion?: string;
+
+  private readonly amiParameterName: string;
+
+  /**
+   * Constructs a new instance of the EcsOptimizedAmi class.
+   */
+  public constructor(props: EksOptimizedAmiProps) {
+    this.nodeType = props && props.nodeType;
+    this.kubernetesVersion = props && props.kubernetesVersion || LATEST_KUBERNETES_VERSION;
+
+    // set the SSM parameter name
+    this.amiParameterName = `/aws/service/eks/optimized-ami/${this.kubernetesVersion}/`
+      + ( this.nodeType === NodeType.NORMAL ? "amazon-linux-2/" : "" )
+      + ( this.nodeType === NodeType.GPU ? " amazon-linux2-gpu/" : "" )
+      + "recommended/image_id";
+  }
+
+  /**
+   * Return the correct image
+   */
+  public getImage(scope: Construct): ec2.MachineImageConfig {
+    const ami = ssm.StringParameter.valueForStringParameter(scope, this.amiParameterName);
+    return {
+      imageId: ami,
+      osType: ec2.OperatingSystemType.LINUX
+    };
+  }
+}
+
+const LATEST_KUBERNETES_VERSION = '1.13';
+
+/**
+ * Whether the worker nodes should support GPU or just normal instances
+ */
+export enum NodeType {
+  /**
+   * Normal instances
+   */
+  NORMAL = 'Normal',
```

Comment:
Maybe “standard”. Normal sounds odd.

Reply:
I'm not a fan of having an enum key be different from its value, but I can change it if you feel strongly about it

Reply:
I just realized this wasn't a CloudFormation value. I'll definitely update it then

Reply:
Done